examples/step-40/doc/intro.dox

```diff
@@ -93,6 +94,71 @@ It is probably worthwhile reading it for background information on how
+keep at most a small multiple of one $N$th of the triangulation.
+Similarly, each process can only hold a small multipe of one $N$th of
+each solution or right hand side vector, and of the system matrix.
+
+To this end, deal.II has acquired interfaces to a number of packages
+that provide these kind of distributed linear algebra data structures.
+More specifically, deal.II comes with a number of "sub-packages" that
+all provide vector, matrix, and linear solver classes that are typically
+named the same or very similarly, but live in different namespaces:
+- deal.II's own linear algebra classes. These are what we have been
+  using in step-1 to step-6, for example, along with most of the
+  other programs in the deal.II tutorial. These classes are all not
+  parallel in the sense that they do not use MPI, can not subdivide
+  the data among processes, or work on them on processes that cannot
+  access each other's memory directory. (On the other hand, many of
+  these classes actually use multiple threads internally, to use
+  the multiple processor cores available on today's laptops and
+  work stations.) These classes reside in the top-level
+  `namespace dealii`.
+- Interfaces to the PETSc library's implementations of linear
+  algebra functionality. These are found in namespace `PETScWrappers`.
+  PETSc is a library that has built a large collection of linear algebra,
+  linear solver, nonlinear solver, time stepper, and other functionality
+  that runs on some of the largest machines in the world in parallel,
+  using MPI.
+- The classes in the TrilinosWrapper namespace wrap functionality
+  provided by the Trilinos project. Trilinos is, in many regards, similar
+  in functionality to what PETSc provides, but it does so in a very different
+  way (namely, as a bunch of independent and loosely coupled sub-projects,
+  rather than as a single library). This nothwithstanding, the classes
+  deal.II provides in namepace TrilinosWrappers are very similar to the
+  ones in namespace PETScWrappers. Trilinos, like PETSc, is run on some of
+  the biggest machines in the world.
+- The classes in namespace TrilinosWrappers are generally written for the
+  functionality Trilinos provides via its "Epetra" collection of vector
+  and matrix classes, and everything that builds on it. Epetra has been
+  slated for removal since the early 2020s (and may already have been removed
+  by the time you read this), with replacements provided in the "Tpetra"
+  collection. Tpetra replaces the old classes by ones that are templatized
+  on the type of objects it stores (thus the "T" at the beginning of the
+  name; "petra" being the word for "rock" or "foundation" since the rest of
+  Trilinos builds on these packages). deal.II is wrapping the Tpetra classes
+  in namespace LinearAlgebra::TpetraWrappers, but this process is still
+  ongoing and may be available in a version of deal.II after release 9.5.
```

Comment:
What do you think about adding a short extension like this:
```suggestion
  ongoing and may be available in a version of deal.II after release 9.5.
- deal.II also implements a subset of parallel linear algebra functionality
  internally through the LinearAlgebra::distributed::Vector class (compatible
  with most Trilinos classes) and matrix-free linear solvers, which are run
  on some of the biggest machines in the world as well. For further details, see
  the step-37 tutorial program.
```

Reply:
Ah yes, good point. I meant to mention this and forgot about it.